test.sh

```diff
@@ -83,13 +86,19 @@ assert 10 '{ i=0; while(i<10) { i=i+1; } return i; }'
 assert 55 '{ i=0; j=0; for (i=0; i<=10; i=i+1) j=i+j; return j; }'
 
 assert 3 '{ x=3; return *&x; }'
-assert 3 '{ x=3; y=&x; z=&y; return **z; }'
-assert 5 '{ x=3; y=5; return *(&x+8); }'
-assert 7 '{ x=3; y=5; z=7; return *(&y+8); }'
-assert 3 '{ x=3; y=5; return *(&y-8); }'
+assert 3 '{ x=3; y=&x; z=&y; return **z;}'
+assert 5 '{ x=3; y=5; return *(&x+1);}'
+assert 7 '{ x=3; y=5; z=7; return *(&y+1);}'
+assert 3 '{ x=3; y=5; return *(&y-1); }'
+assert 5 '{ x=3; y=&x; *y=5; return x; }'
+assert 7 '{ x=3; y=5; *(&x+1)=7; return y; }'
+assert 7 '{ x=3; y=5; *(&y-1)=7; return x; }'
+
+assert 3 '{ x = 2; return x + 1; }'
+assert 2 '{ x = 1; y = 2; z = &x; return *(z+1); }'
 assert 5 '{ x=3; y=&x; *y=5; return x; }'
-assert 7 '{ x=3; y=5; *(&x+8)=7; return y; }'
-assert 7 '{ x=3; y=5; *(&y-8)=7; return x; }'
 
+assert 1 '{ a=1;b=2; return &b-&a; }'
+assert 5 '{ x=3; return &x+2-&x+3; }'
 
```

Comment:
Add this suggested test that I left in the other comment:
`assert 5 '{ a=3; b=5; c=&a; d=&c; return *((*d) + 1);}'`